test/FunctionalTests/BuilderTests/BuilderFilteringWebTests.cs

```diff
@@ -1,77 +1,23 @@
 using System.Linq;
-using System.Threading.Tasks;
-using Microsoft.AspNetCore.Server.IntegrationTesting;
-using Microsoft.AspNetCore.Testing.xunit;
+using System.Net.Http;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace EntropyTests.BuilderTests
 {
-    public class BuilderFilteringWebTests
+    public class BuilderFilteringWebTests : E2ETestBase
     {
-        private const string SiteName = "Builder.Filtering.Web";
-        private readonly ITestOutputHelper _output;
         public BuilderFilteringWebTests(ITestOutputHelper output)
+            : base(output, "Builder.Filtering.Web", 5500)
```

Comment:
There is an API for it in Hosting (https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/TestUriHelper.cs#L35), but I'm not sure how you'd make it work without locking. These tests run in parallel (and I'd like to keep it that way).